validphys2/src/validphys/photon/compute.py

```diff
@@ -247,3 +262,39 @@ def generate_errors(self, replica):
         u, s, _ = np.linalg.svd(self.error_matrix, full_matrices=False)
         errors = u @ (s * rng.normal(size=7))
         return errors
+    
+    def load_photon(self):
+      """Load the photon resource using the Loader class."""
+      path_to_photon = loader.check_photonQED(self.theoryid.id, self.luxpdfset._name)
+      log.info(f"Loading photon QED set from {path_to_photon}")
+
+      interpolator = []
+      integral = []
+
+      # Load the needed replicas
+      for replica in self.replicas:
+          # As input replica for the photon computation we take the MOD of the luxset_members to
+          # avoid failing due to limited number of replicas in the luxset
+          photonreplica = (replica % self.luxpdfset_members) or self.luxpdfset_members
+
+          photon_array = np.load(path_to_photon / f"replica_{photonreplica}.npz")["photon_array"]
+          interpolator.append(interp1d(XGRID, photon_array, fill_value="extrapolate", kind="cubic"))
+          integral.append(trapezoid(photon_array, XGRID))
+      
+      self.interpolator = interpolator
+      self.integral = np.stack(integral, axis=-1)
+      return
+    
+def compute_photon(theoryid, fiatlux):
+    """Function to compute the photon PDF set.""" 
+    luxset = fiatlux['luxset'].load()
+    force_compute = fiatlux.get('compute_in_setupfit', False)
+    replicas = list(range(1, luxset.n_members))
+
+    # Return None and avoid pickling issues with the phoyon class.
+    def wrapper_fn(replica, theoryid, fiatlux, force_compute):
+        Photon(theoryid, fiatlux, replicas=[replica], save_to_disk=True, force_computation=force_compute)
+        return None
+    
+    log.info(f"Starting computation of the photon using {effective_n_jobs(-1)} effective cores...")
+    _ = Parallel(n_jobs=-1)(delayed(wrapper_fn)(replica=replica, theoryid=theoryid, fiatlux=fiatlux, force_compute=force_compute) for replica in replicas)
```

Comment:
Don't do `n_jobs=-1` but use the `maxcores` flag from the runcard. `n_jobs=-1` can be horrible if the cluster is not well configured (e.g., if you are allowed to see other cores in your node even if you didn't ask for them)